lib/metadata/schema-parser.js

```diff
@@ -1009,37 +1009,24 @@ function getTokenToReplicaSimpleMapper(replicationFactor) {
  * @private
  */
 function getTokenToReplicaNetworkMapper(replicationFactors) {
-  //                A(1)
+  //                A(DC1)
   //
-  //           H         B(2)
+  //           H         B(DC2)
   //                |
-  //      G       --+--       C(1)
+  //      G       --+--       C(DC1)
   //                |
-  //           F         D(2)
+  //           F         D(DC2)
   //
-  //                E(1)
+  //                E(DC1)
   return (function tokenNetworkStrategy(tokenizer, ring, primaryReplicas, datacenters) {
-    function isDoneForToken(replicasByDc) {
-      for (var dc in replicationFactors) {
-        if (!replicationFactors.hasOwnProperty(dc)) {
-          continue;
-        }
-        var rf = Math.min(parseInt(replicationFactors[dc], 10), datacenters[dc]);
-        if (replicasByDc[dc] < rf) {
-          return false;
-        }
-      }
-      return true;
-    }
-    //For each token
-    //Get an Array of tokens
-    //Checking that there aren't more tokens per dc than specified by the replication factors
     var replicas = {};
     for (var i = 0; i < ring.length; i++) {
       var token = ring[i];
       var key = tokenizer.stringify(token);
       var tokenReplicas = [];
```

Comment:
That is what I have done, but it's not quite behaving as I like, diving into a debugger to see what I'm missing.  It is likely i'm doing something wrong.  I'm attempting to simulate 100 hosts with 256 vnodes each and then ensure that the replicas returned are as expected, this isn't in the best shape but here's what i currently got:

```javascript

    it('should run expediently with many replicas and 0 nodes in DC.', function (done) {
      var cc = getControlConnectionForRows([{
        'keyspace_name': 'dummy',
        'strategy_class': 'NetworkTopologyStrategy',
        'strategy_options': '{"dc1": "3", "dc2": "3"}'
      }]);
      var options = clientOptions.extend({}, helper.baseOptions);
      var metadata = new Metadata(options, cc);
      metadata.tokenizer = getTokenizer();
      var racksDc1 = new utils.HashSet();
      racksDc1.add('dc1_r1');
      var racksDc2 = new utils.HashSet();
      racksDc2.add('dc2_r1');
      metadata.datacenters = {
        'dc1': { hostLength: 100, racks: racksDc1 },
        'dc2': { hostLength: 0, racks: racksDc2 }};

      metadata.ring = [];
      // create ring and replicas
      metadata.primaryReplicas = {};
      for (var r = 0; r < 100; r++) {
        var h = new Host(r.toString(), 2, options);
        if (r == 0) {
          h.datacenter = 'dc1';
          h.rack = 'dc1_r1';
        } else {
          h.datacenter = 'dc2';
          h.rack = 'dc2_r1';
        }
        // 256 vnodes per replica.
        for (var v = 0; v < 256; v++) {
          var token = (v * 256) + r;
          metadata.ring.push(token);
          metadata.primaryReplicas[token.toString()] = h;
        }
      }
      // sort the ring so the tokens are in order (this is done in metadata buildTokens, but it accounts for
      // partitioner which we don't need to use here).
      metadata.ring.sort(function (a, b) { return a - b });

      metadata.log = helper.noop;
      // Get the replicas of 0.  Since token 0 is a replica in DC1 and DC1 only has 0, it should return 0 and then
      // fallback on 1 and 2.
      metadata.refreshKeyspaces(function () {
        var replicas = metadata.getReplicas('dummy', new Buffer([0]));
        assert.ok(replicas);
        assert.deepEqual(replicas.map(getAddress), ['0', '0', '0']);
        // Get the replicas on
        metadata.refreshKeyspaces(function () {
          var replicas = metadata.getReplicas('dummy', new Buffer([7000]));
          assert.ok(replicas);
          assert.deepEqual(replicas.map(getAddress), ['2', '3', '4']);
          done();
        });
      });
```

I see in your tests that you are getting the right amount of replicas back, so it's incredibly likely i'm missing something :)